src/test/java/InfrastructureManager/AdvantEdge/AdvantEdgeClientTests.java

```diff
@@ -0,0 +1,78 @@
+package InfrastructureManager.AdvantEdge;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.File;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options;
+
+
+public class AdvantEdgeClientTests {
+    private final AdvEClient client = new AdvEClient();
+    private final String scenarioName = "dummy-test";
+
+    @Rule //Mock server on port 80
+    public WireMockRule rule = new WireMockRule(options().port(80),false);
+
+
+    @Test
+    public void addScenarioRequestTest() {
+        String path = "/scenarios/" + scenarioName;
+        String scenarioPath = "src/test/resources/AdvantEdge/dummy-test.json";
+        client.out("advantEdge create " + scenarioName + " " + scenarioPath);
+        verify(postRequestedFor(urlEqualTo(path))
+                .withRequestBody(
+                        matchingJsonPath("$.name", equalTo(scenarioName))
+                ));
+    }
+
+    @Test // Checks if adding the scenario as a YAML file, sends a JSON in the request
+    public void YAML2JSONParserTest() {
```

Comment:
since this test is more about the functionality of sending request with a YAML file and parsing isn't really the responsibility of AdvantEdgeClient, I think it'd be more clear if we name it as such
```suggestion
    public void addYAMLScenarioRequestTest() {
```